### PR TITLE
fix(group): resolve JVM workspace Maven and Gradle artifact coordinates

### LINKED ISSUE #3106: java-workspace-extractor skips repos sharing a parent POM as "duplicate artifact" — breaks manifest cross-links
### Area

gitnexus (CLI / core / indexing / MCP server)

### Summary

`java-workspace-extractor` skips repositories that share a common Maven parent POM, reporting them as "duplicate artifact", so `group sync` builds zero `manifest` cross-links even though every `pom.xml` is correctly parsed and artifact coordinates are recognized.

### Context

A group of independent Java/Maven git repositories that all inherit from the same `<parent>` POM — the standard layout for a portfolio of related services / shared libraries under one organization (e.g. multiple repos inheriting `spring-boot-starter-parent`). Each repo is indexed independently and added to a group. `workspace_deps` is enabled in `group.yaml`. The extractor clearly reads each repo's `groupId:artifactId` from its `pom.xml`, but then skips nearly every repo because they share the inherited parent artifact id.

This makes the `workspace_deps` cross-linking feature effectively unusable for this extremely common Maven shape.

### Expected behavior

- A shared parent POM is inherited, not the repo's own deliverable artifact. Repositories that merely share an inherited `<parent>` should not be treated as duplicates of each other or of the parent.
- The extractor should build cross-links from a repo's `<dependency>` declarations to the repo that publishes that dependency artifact — even when multiple repos share the same parent.
- In a group where `service-a` and `service-b` both declare `<dependency>` on `com.example:shared-lib`, each should get a `manifest` cross-link to the `shared-lib` repo, regardless of all of them inheriting `com.example:parent`.

### Actual behavior

`group sync --verbose` recognizes every artifact but skips all but the first repo per shared artifact id:

```
[java-workspace-extractor] duplicate artifact "com.example:parent" in "service-a" and "parent" — skipping "service-a"
[java-workspace-extractor] duplicate artifact "com.example:parent" in "service-b" and "parent" — skipping "service-b"
[java-workspace-extractor] duplicate artifact "com.example:parent" in "shared-lib" and "parent" — skipping "shared-lib"
...
```

Result:

```
Matching:
  exact:     6 cross-links (confidence 1.0)
  manifest:  0 cross-links
  wildcard:  0 cross-links
```

`manifest: 0 cross-links` — the `service-a → shared-lib` and `service-b → shared-lib` Maven dependency edges are never created, because the repos carrying those `<dependency>` declarations were skipped as "duplicate artifact".

### Steps to reproduce

Setup: independent Maven repositories sharing a common parent POM:

- `parent` — root POM, `com.example:parent`, `packaging: pom`
- `shared-lib` — `<parent>` is `com.example:parent`; publishes `com.example:shared-lib`
- `service-a` — `<parent>` is `com.example:parent`; `<dependency>` on `com.example:shared-lib`
- `service-b` — `<parent>` is `com.example:parent`; `<dependency>` on `com.example:shared-lib`
- (several more repos inheriting the same parent)

1. Index each repo independently:
   ```bash
   gitnexus analyze <repo> --force --index-only
   ```
2. Create the group and add members:
   ```bash
   gitnexus group create mygroup
   gitnexus group add mygroup parent parent
   gitnexus group add mygroup shared-lib shared-lib
   gitnexus group add mygroup service-a service-a
   gitnexus group add mygroup service-b service-b
   ```
3. Enable workspace dependency detection in `group.yaml` (off by default):
   ```yaml
   detect:
     workspace_deps: true
   ```
4. Sync:
   ```bash
   gitnexus group sync mygroup --verbose
   ```
5. Observe the "duplicate artifact ... skipping" log lines and `manifest: 0 cross-links`.

### Environment

- OS: macOS (arm64)
- Node: v24.15.0
- Commit / version: gitnexus 1.6.10

### Logs / screenshots

See the "duplicate artifact ... skipping" lines and the `manifest: 0 cross-links` summary quoted in *Actual behavior* above.

### Notes

- `pom.xml` files ARE indexed as `File` nodes and their artifact coordinates ARE parsed (the log shows the extractor reading `groupId:artifactId` correctly) — the issue is purely the duplicate-artifact skip logic, not parsing.
- The `ContractType` enum already includes `'lib'`, and `extractJavaWorkspaceLinks` + `ManifestExtractor` are wired up — so this looks like a logic gap in duplicate handling rather than a missing feature.
- `group query` (federated execution-flow search) works across repos and is a usable fallback for discovery, but it does not represent the Maven dependency edges as cross-links.

### PR DESCRIPTION
**Pull request by magyargergo**
## Summary

`group sync` dropped sibling JVM modules because workspace identity and deps were parsed incorrectly. This PR keeps the extractor **static** (never runs Maven or Gradle) and makes coordinates match what the manifests actually declare.

- **Maven:** parse each `pom.xml` with `fast-xml-parser`. Child `groupId` comes from the project, else `<parent>`; `artifactId` is only the project's own element. Dependencies are `project.dependencies` only (not BOM / profiles / plugin classpath). Duplicate `groupId:artifactId` keys are still skipped.
- **Gradle:** literals plus sidecars (`gradle.properties` `group=`, `settings.gradle(.kts)` `rootProject.name`, default `gradle/libs.versions.toml`). Groovy/Kotlin `group`, `allprojects { group = … }`, named `group`/`name` in either order, catalog aliases (including underscores and `libs.foo.get()`), bundles, and type-safe `projects.sharedLib` → kebab `artifactId`. If `pom.xml` exists, Gradle is not read.
- Cover the #3106 parent-POM layout in extractor + `syncGroup` tests so `manifest` cross-links are non-zero.

Closes #3106

## Test plan

- [x] `cd gitnexus && npx vitest run test/unit/group/java-workspace-extractor.test.ts test/unit/group/sync.test.ts`
- [x] `cd gitnexus && npx vitest run test/unit/group/*workspace-extractor.test.ts test/unit/group/manifest-extractor.test.ts test/unit/group/sync.test.ts`
- [x] `cd gitnexus && npm run build`
- [ ] Confirm CI unit tests

---

## Summary

*A focused, low-risk change concentrated in the Java workspace extraction path, with supporting dependency metadata and unit test updates. Its reach is shallow and remains within the group extraction area.*

**🟢 LOW blast radius. A JVM workspace artifact resolution change in `gitnexus/src/core/group/extractors/java-workspace-extractor.ts`, reaching one direct dependent and one second-hop dependent.**

The change is centered on Maven and Gradle coordinate parsing through `parseJavaManifest`, `parsePom`, `parseGradle`, and `extractJavaWorkspaceLinks`, alongside Java workspace discovery helpers such as `deriveBasePackage`, `scanJavaImports`, and `findJavaFiles`. The primary impact lands in the `Extractors` module, with a smaller connection to the `Group` module.

Review `gitnexus/src/core/group/extractors/java-workspace-extractor.ts` first, then the related coverage in `gitnexus/test/unit/group/java-workspace-extractor.test.ts` and `gitnexus/test/unit/group/sync.test.ts`. The remaining changed files are `gitnexus/package.json` and `gitnexus/package-lock.json`.

_Added by GitNexus for PR #3108. Edit freely — this block is replaced on the next review, everything above it is left untouched._

### CODE REVIEW

File: gitnexus/src/core/group/extractors/java-workspace-extractor.ts [hunk 1]
```diff
@@ -44,59 +61,275 @@ async function parseJavaManifest(
...
+      const members = [...assignment[2].matchAll(/['"]([^'"]+)['"]/g)].map((match) => match[1]);
+      for (const accessor of catalogAccessors(assignment[1])) bundles.set(accessor, members);
     }
   }
 
+  return { libraries, bundles };
+}
+
+const GRADLE_GROUP_PATTERNS = [
+  /(?:^|\n)\s*(?:rootProject\.)?group\s*=\s*['"]([^'"]+)['"]/,
```
[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts]: > valid Gradle such as `allprojects { group = "com.acme" }` is not recognized

Fixed in 210762990 — `parseGradleGroup` now treats `{` and `;` as group-assignment boundaries, so `allprojects { group = "com.acme" }` is picked up without requiring a `gradle.properties` sidecar.

**Comment by magyargergo** [comment]
> Groovy named coordinates with `name` first are not extracted … a declaration such as `implementation name: 'shared', group: 'com.acme'` produces no dependency entry

Fixed in 210762990 — Groovy map notation now accepts both `group:`/`name:` orderings (same as the Kotlin named-arg forms). Covered by the `named-first` workspace fixture.

File: gitnexus/src/core/group/extractors/java-workspace-extractor.ts [hunk 2]
```diff
@@ -44,59 +61,282 @@ async function parseJavaManifest(
...
+
+const GRADLE_GROUP_PATTERNS = [
+  /(?:^|[\n{;])\s*(?:rootProject\.)?group\s*=\s*['"]([^'"]+)['"]/,
+  /(?:^|[\n{;])\s*group\s+['"]([^'"]+)['"]/,
+];
+
+const GRADLE_COORD_CONFIGS =
+  'implementation|api|compileOnly|runtimeOnly|testImplementation|testApi|testCompileOnly|compile|kapt|ksp|commonMainImplementation|commonMainApi';
+
+const CATALOG_ALIAS = '([A-Za-z0-9]+(?:\\.[A-Za-z0-9]+)*)(?:\\.get\\(\\)|\\.asProvider\\(\\))?';
```
[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts]: > a valid catalog entry such as `foo_bar = { module = "com.acme:foo" }` is not resolved from `libs.foo_bar`

Fixed in e162f0c72 — `CATALOG_ALIAS` now allows `_` in accessor segments, so `implementation(libs.foo_bar)` looks up the stored catalog key. Covered by a dedicated extractor test.

File: gitnexus/src/core/group/extractors/java-workspace-extractor.ts [hunk 3]
```diff
@@ -44,59 +61,282 @@ async function parseJavaManifest(
...
+    const groupId = xmlChildText(depNode, 'groupId');
+    const artifactId = xmlChildText(depNode, 'artifactId');
+    if (groupId && artifactId) deps.push(`${groupId}:${artifactId}`);
+  }
+}
+
+function parsePom(content: string): { groupId: string; artifactId: string; deps: string[] } | null {
+  let parsed: unknown;
+  try {
+    parsed = parseSourceSafe(pomParser, content);
```
[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts]: > The added parser call is `parsed = parseSourceSafe(pomParser, content)`, but the complete PR-head file contains no declaration or import for `parseSourceSafe`

Fixed in e162f0c72 — restored `pomParser.parse(content)`. `parseSourceSafe` is the tree-sitter helper and was introduced by `/autofix`; it is the wrong API for `fast-xml-parser`.

File: gitnexus/src/core/group/extractors/java-workspace-extractor.ts [hunk 4]
```diff
@@ -44,59 +61,282 @@ async function parseJavaManifest(
...
+  return sidecars;
+}
+
+function catalogAccessors(alias: string): string[] {
+  const dotted = alias.replace(/[-_]/g, '.');
+  const camel = alias.replace(/[-_]+([A-Za-z0-9])/g, (_, char: string) => char.toUpperCase());
+  return [...new Set([alias, dotted, camel])];
+}
+
+function projectAccessorToArtifactId(accessor: string): string {
```
[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts:114]: > `projects.foo.bar` is the type-safe accessor for the Gradle project path `:foo:bar`, but the new helper splits on `.` and keeps only `bar`

Not changing this. Workspace identity is `groupId:artifactId` from `rootProject.name` / the repo directory, so the type-safe accessor is converted to that leaf artifact id (`projects.sharedLib` → `shared-lib`, `projects.foo.bar` → `bar`). Recording `group:foo:bar` like `project(":foo:bar")` would not match any discovered project. The `project()` string form is a separate, pre-existing path and is not the matching key this helper is meant to produce.

File: gitnexus/src/core/group/extractors/java-workspace-extractor.ts [hunk 5]
```diff
@@ -44,59 +61,282 @@ async function parseJavaManifest(
...
+    const groupId = xmlChildText(depNode, 'groupId');
+    const artifactId = xmlChildText(depNode, 'artifactId');
+    if (groupId && artifactId) deps.push(`${groupId}:${artifactId}`);
+  }
+}
+
+function parsePom(content: string): { groupId: string; artifactId: string; deps: string[] } | null {
+  let parsed: unknown;
+  try {
+    parsed = parseSourceSafe(pomParser, content);
```
[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts]: > The new POM parser calls `parseSourceSafe` at line 260, but this file's imports (lines 1–8) do not import it

Fixed in 327ed685b — back to `pomParser.parse(content)`. The `parseSourceSafe` call was inserted by the `/autofix` run (the `gitnexus/require-safe-parse` rule auto-rewrites any `<receiver>.parse(expr)` under `src/core/`), never by a hand edit, so no import was ever intended. The call site now carries an `eslint-disable-next-line` with the reason.

[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts]: > `XMLParser.parse` expects XML text, not a tree-sitter input callback, so the new call at line 260 cannot safely parse larger POMs; the catch then silently treats that POM as invalid

Confirmed and fixed in 327ed685b. Measured on a 35 195-char POM with 400 `<dependency>` entries: `pomParser.parse` returns `artifactId: svc` with 400 deps, while `parseSourceSafe(pomParser, …)` returns an empty document (`project` undefined) — and it does not throw, so the `catch` never fires and the POM is dropped silently, exactly as described.

`parsePom` now calls `pomParser.parse(content)` behind an `eslint-disable-next-line gitnexus/require-safe-parse` plus a comment, so the autofix rule cannot reintroduce the rewrite.

File: gitnexus/src/core/group/extractors/java-workspace-extractor.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import fs from 'node:fs/promises';
 import path from 'node:path';
+import { XMLParser } from 'fast-xml-parser';
```
[the remark this answers is not in the document]
Comment (magyargergo) [reply gitnexus/src/core/group/extractors/java-workspace-extractor.ts:3]: > neither package.json nor the lockfile is changed in the PR to declare or install it

Not addressing — the dependency is declared. `gitnexus/package.json` line 69 has `"fast-xml-parser": "^5.11.1"` and `gitnexus/package-lock.json` carries both the root entry (line 23) and the resolved `node_modules/fast-xml-parser` 5.11.1 package (line 3065). Both files are in this PR's diff.
